File: 백준/Gold/1931. 회의실 배정/회의실 배정.swift
// [Int] N (immutable): An Integer that represents number of meeting(range: 1 ≤ N ≤ 100,000)
if let input: String = readLine(), let N: Int = Int(input) {
    // [Int] lastEnd (mutable): An Integer that stores last meeting time
    var lastEnd: Int = 0
    
    // [int] count (mutable): An Integer that stores the number of meetings
    var count: Int = 0
    
    // [Array<Array<Int>>] arrayOfTimes (mutable): 2-Dimensional arrya of integers that 
    // stores start and end time of each meetings
    var arrayOfTimes = [[Int]]()
    
    for _ in 0..<N {
        let times: [Int] = (readLine() ?? "1 1").split(separator: " ").compactMap { Int($0) }
        arrayOfTimes.append( times )
    }
    
    // Sort by end time of meetings
    arrayOfTimes.sort { $0[1] == $1[1] ? $0[0] < $1[0] : $0[1] < $1[1] }
    
    for times in arrayOfTimes {
        // [Int] start, end (immutable): Start and end times of the now meeting
        let (start, end): (Int, Int) = (times[0], times[1])
        
        // Update count and lastEnd when start is later than or equal to lastEnd
        if (start >= lastEnd) {
            count += 1
            lastEnd = end
        }
    }
    print(count)
}

File: 백준/Gold/1931. 회의실 배정/회의실 배정.py
import sys

input = sys.stdin.readline

# [int] N: An Integer that represents number of meeting(range: 1 ≤ N ≤ 100,000)
N = int(input())

# [int] lastEnd: An Integer that stores last meeting time
lastEnd = 0

# [int] count: An Integer that stores the number of meetings
count = 0

# [list(tuple)] listOfTuple: List of tuple of integers that 
# stores start and end time of each meetings
listOfTuple = list()

for _ in range(N):
    # [int] start, end: Start and end times of the now meeting
    start, end = map(int, input().split())
    listOfTuple.append( (start, end) )

# Sort by end time of meetings
listOfTuple.sort(key= lambda x: (x[1], x[0]))

for start, end in listOfTuple:
    # Update count and lastEnd when start is later than or equal to lastEnd
    if (start >= lastEnd):
        count += 1
        lastEnd = end

print(count)